selfservice/hook/web_hook.go

```diff
@@ -412,10 +412,10 @@ func parseWebhookResponse(resp *http.Response, id *identity.Identity) (err error
 	}
 
 	if resp.StatusCode == http.StatusOK {
+		type localIdentity identity.Identity
 		var hookResponse struct {
-			Identity *identity.Identity `json:"identity"`
+			Identity *localIdentity `json:"identity"`
```

Comment:
This exposes credentials though. Please use the proper types from the identity package, in this case `identity.WithAdminMetadataInJSON`

Reply:
No it doesn't, please have a look at what's being done with `hookResponse` below.